### ISSUE TITLE
[P1][packaging][security] SHA256SUMS do wheelhouse não cobre as 10 células cp314t — 10 de 10 sem checksum, verificação offline impossível

### ISSUE THREAD

**Issue by FabioLeitao** [DataBoar/data-boar#1410]
## Problema

O `SHA256SUMS` da release [`wheelhouse-x86-64-v1-2026-07-29`](https://github.com/DataBoar/data-boar-site/releases/tag/wheelhouse-x86-64-v1-2026-07-29) **não cobre nenhuma das 10 células `cp314t`**.

Medido em 2026-07-30, baixando os assets e cruzando com o arquivo de checksums:

```
⚠ sem entrada: boar_fast_filter-0.1.0-cp314-cp314t-manylinux_2_28_x86_64.whl
⚠ sem entrada: boar_fast_filter-0.1.0-cp314-cp314t-musllinux_1_2_x86_64.whl
⚠ sem entrada: numpy-2.5.1-cp314-cp314t-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl
⚠ sem entrada: numpy-2.5.1-cp314-cp314t-musllinux_1_2_x86_64.whl
⚠ sem entrada: pandas-3.0.5-cp314-cp314t-manylinux_2_24_x86_64.manylinux_2_28_x86_64.whl
⚠ sem entrada: pandas-3.0.5-cp314-cp314t-musllinux_1_2_x86_64.whl
⚠ sem entrada: scikit_learn-1.9.0-cp314-cp314t-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl
⚠ sem entrada: scikit_learn-1.9.0-cp314-cp314t-musllinux_1_2_x86_64.whl
⚠ sem entrada: scipy-1.18.0-cp314-cp314t-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl
⚠ sem entrada: scipy-1.18.0-cp314-cp314t-musllinux_1_2_x86_64.whl

10 de 10 sem cobertura.
```

O formato do arquivo está correto e as 46 células originais estão lá — o problema é que as **10 células `cp314t` foram acrescentadas em 2026-07-30, depois** de o `SHA256SUMS` ter sido gerado, e ele não foi regerado.

Ver `PLAN_WHEELHOUSE_DISTRIBUTION.md`, seção *"Hosted release — x86-64-v1 (verified 2026-07-29)"*: `56` assets = **46** células GIL/v1 originais + **10× `cp314t`** (adicionadas 2026-07-30).

## Por que importa mais aqui do que em outro projeto

Este é o **wheelhouse** — o insumo da camada 1, a que define se o scan devolve 26 achados ou 13. Um artefato de supply chain publicado sem checksum, num produto de compliance, é o tipo de coisa que o próprio produto existe para encontrar no cliente.

E não é hipotético: o `README.md` da release documenta verificação por checksum como parte do fluxo de instalação offline. Para as 10 células novas, essa instrução **não tem como ser cumprida**.

O plano também registra que a prova offline de 2026-07-29 (*"Operator re-downloaded; checksums matched"*) foi feita **antes** das células `cp314t` entrarem — então elas nunca passaram por essa verificação.

## Proposta

1. **Regerar o `SHA256SUMS`** cobrindo os 56 assets e substituir na release.
2. **Gate no fluxo de publicação:** falhar se a contagem de linhas do `SHA256SUMS` ≠ número de assets `.whl` da release. É o que impede a divergência de voltar quando a próxima célula for acrescentada — mesmo padrão de raciocínio do smoke de extras (#1401).
3. **Reverificar offline** as 10 células, como foi feito com as 46.

## Aceite

- [ ] `SHA256SUMS` cobre os **56** assets da release `wheelhouse-x86-64-v1-2026-07-29`
- [ ] Verificação offline das 10 células `cp314t` executada e registrada
- [ ] Gate de publicação: contagem de checksums = contagem de wheels, falha se divergir
- [ ] `PLAN_WHEELHOUSE_DISTRIBUTION.md` atualizado com a data real da prova offline das células `cp314t`
- [ ] Conferir se a release anterior (`wheelhouse-2026-07-12`) tem a mesma lacuna

## Nota de descoberta

Encontrado ao usar o wheelhouse para resolver a célula **musl** do pacote nativo (#1403): das 9 dependências compiladas testadas, **apenas `scikit-learn` não tem wheel `musllinux` para `cp314t` na PyPI** — e o wheelhouse tem exatamente esse arquivo. O `--find-links` resolveu a instalação e a verificação passou (`musl` · `3.14.6` · GIL desligado no boot **e** após a stack completa · `0 .so` em sqlalchemy · numpy 2.5.1 / pandas 3.0.5 / sklearn 1.9.0).

Ou seja: **o wheelhouse funcionou e provou seu valor** — só não dá para verificar a integridade do que ele entregou.

Relacionadas: #1182 / `PLAN_WHEELHOUSE_DISTRIBUTION.md` · #1403 · #1405

**Comment by FabioLeitao** [comment]
Executor update (2026-08-17):

1. Regenerated `SHA256SUMS` on `wheelhouse-x86-64-v1-2026-07-29` — **54** lines (covers all `.whl`, including **10× cp314t**). Prior file had 41 lines / 0 cp314t.
2. Offline `sha256sum -c`: **54/54 OK**, **10/10 cp314t OK**.
3. Gate: `scripts/wheelhouse/verify_release_sha256sums.sh` + CI job on `wheelhouse-recipe.yml` (PR pending).
4. Plan updated with proof date **2026-08-17**.
5. Seed `wheelhouse-2026-07-12`: 1 wheel, **no** `SHA256SUMS` asset (different gap — not cp314t lag).

PR will close this issue when merged.
